src/Transport/CliSnmpClient.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\PhpSnmp\Transport;
+
+use SimPod\PhpSnmp\Exception\EndOfMibReached;
+use SimPod\PhpSnmp\Exception\GeneralException;
+use SimPod\PhpSnmp\Exception\InvalidVersionProvided;
+use SimPod\PhpSnmp\Exception\NoSuchInstanceExists;
+use SimPod\PhpSnmp\Exception\NoSuchObjectExists;
+use SimPod\PhpSnmp\Transport\Cli\ProcessExecutor;
+use SimPod\PhpSnmp\Transport\Cli\SymfonyProcessProcessExecutor;
+use function array_merge;
+use function count;
+use function explode;
+use function in_array;
+use function Safe\preg_match;
+use function Safe\preg_split;
+use function strpos;
+
+final class CliSnmpClient implements SnmpClient
+{
+    /** @var ProcessExecutor */
+    private $processExecutor;
+
+    /** @var string[] */
+    private $processArgs;
+
+    /** @var bool */
+    private $useBulk;
+
+    public function __construct(
+        string $host = '127.0.0.1',
+        string $community = 'public',
+        int $timeout = 1,
+        int $retries = 3,
+        string $version = '2c',
+        ?ProcessExecutor $processExecutor = null
+    ) {
+        if (! in_array($version, ['1', '2c'], true)) {
+            throw InvalidVersionProvided::new($version);
+        }
+
+        $this->processExecutor = $processExecutor ?? new SymfonyProcessProcessExecutor(120);
+        $this->processArgs     = [
+            '-ObenU',
+            '-t',
+            (string) $timeout,
+            '-r',
+            (string) $retries,
+            '-v',
+            $version,
+            '-c',
+            $community,
+            $host,
+        ];
+        $this->useBulk         = $version === '2c';
```

Comment:
I'd drop useBulk, store version instead

Reply:
no need as I just need the information whether to use bulk operation or not, no need for the version to be stored...

Reply:
yeah, I know, but why? :D I don't need the version, I just need that info ... I think it's fine to just narrow it down in the constructor

Reply:
Personally I would not introduce new property just because of that when I have `version` value already. But it's only my preference.

Reply:
If I used version as property, then yeah of course that would make sense. But I don't have a version property.

Reply:
yup, just sayin i would have chosen version property instead :D